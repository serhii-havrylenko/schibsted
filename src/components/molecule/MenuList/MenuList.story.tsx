import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { StaticRouter } from 'react-router';
import styled from 'styled-components';

import MenuList from './MenuList';

const items = [{ title: 'Home', to: '/home' }, { title: 'FAQ', to: '/faq' }];
const Wrapper = styled.div`
  width: 100%;
  height: 70px;
  background-color: #abcdef;
`;

storiesOf('MenuList', module)
  .addDecorator((story) => (
    <StaticRouter context={{}}>
      <Wrapper>{story()}</Wrapper>
    </StaticRouter>
  ))
  .add('default', () => <MenuList items={items} />);
