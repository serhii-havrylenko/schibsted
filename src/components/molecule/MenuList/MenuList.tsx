import * as React from 'react';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

import MenuItem from '#atom/MenuItem';

export interface MenuItem {
  title: string;
  to: string;
}

export interface MenuList {
  items: MenuItem[];
}

const Wrapper = styled.div`
  display: flex;
  height: 70px;
  flex: 1;

  & > a {
    color: #fff;
    text-decoration: none;
    align-self: center;
  }

  & > *:not(:first-child) {
    margin-left: 10px;
  }
`;

const MenuList: React.SFC<MenuList> = ({ items }) => (
  <Wrapper>
    {items.map(({ to, title }, id) => (
      <NavLink to={to} key={id}>
        <MenuItem>{title}</MenuItem>
      </NavLink>
    ))}
  </Wrapper>
);

export default MenuList;
