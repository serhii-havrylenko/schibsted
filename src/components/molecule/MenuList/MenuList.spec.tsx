import { mount } from 'enzyme';
import * as React from 'react';

import MenuList from './MenuList';

jest.mock('@material-ui/core/Typography');
jest.mock('react-router-dom', () => {
  require('react');
  return {
    NavLink: ({ children }: { children: React.ReactNode }) => (
      <div>{children}</div>
    ),
  };
});

describe('<MenuList /> tests', () => {
  const props = {
    items: [{ title: 'Home', to: '/home' }, { title: 'FAQ', to: '/faq' }],
  };

  test('should match snapshot', () => {
    const tree = mount(<MenuList {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
