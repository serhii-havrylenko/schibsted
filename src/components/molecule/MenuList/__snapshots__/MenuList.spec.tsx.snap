// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuList /> tests should match snapshot 1`] = `
.c1.c1 {
  color: #fff;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 70px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 > a {
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0 > *:not(:first-child) {
  margin-left: 10px;
}

<MenuList
  items={
    Array [
      Object {
        "title": "Home",
        "to": "/home",
      },
      Object {
        "title": "FAQ",
        "to": "/faq",
      },
    ]
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  height: 70px;
  flex: 1;

  & > a {
    color: #fff;
    text-decoration: none;
    align-self: center;
  }

  & > *:not(:first-child) {
    margin-left: 10px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <NavLink
          key="0"
          to="/home"
        >
          <div>
            <Styled(MenuItem)>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": true,
                      "lastClassName": "c1",
                      "rules": Array [
                        "
  && {
    color: #fff;
  }
",
                      ],
                    },
                    "displayName": "Styled(MenuItem)",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": [Function],
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <MenuItem
                  className="c1"
                >
                  <WithStyles(Typography)
                    className="c1"
                    color="inherit"
                    noWrap={false}
                    variant="h6"
                  />
                </MenuItem>
              </StyledComponent>
            </Styled(MenuItem)>
          </div>
        </NavLink>
        <NavLink
          key="1"
          to="/faq"
        >
          <div>
            <Styled(MenuItem)>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": true,
                      "lastClassName": "c1",
                      "rules": Array [
                        "
  && {
    color: #fff;
  }
",
                      ],
                    },
                    "displayName": "Styled(MenuItem)",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": [Function],
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <MenuItem
                  className="c1"
                >
                  <WithStyles(Typography)
                    className="c1"
                    color="inherit"
                    noWrap={false}
                    variant="h6"
                  />
                </MenuItem>
              </StyledComponent>
            </Styled(MenuItem)>
          </div>
        </NavLink>
      </div>
    </StyledComponent>
  </styled.div>
</MenuList>
`;
